tools/webpack-plugin/src/BacktracePlugin.ts

```diff
@@ -1,19 +1,77 @@
-import webpack from 'webpack';
-import { BacktracePluginV4 } from './BacktracePluginV4';
-import { BacktracePluginV5 } from './BacktracePluginV5';
-
-let BacktracePlugin: typeof BacktracePluginV4 | typeof BacktracePluginV5;
-
-const version = process.env.WEBPACK_VERSION ?? webpack.version[0];
-switch (version) {
-    case '4':
-        BacktracePlugin = BacktracePluginV4;
-        break;
-    case '5':
-        BacktracePlugin = BacktracePluginV5;
-        break;
-    default:
-        throw new Error(`Webpack version ${version} is not supported.`);
-}
+import { DebugIdGenerator, SourceMapUploader, SourceProcessor } from '@backtrace/sourcemap-tools';
+import path from 'path';
+import webpack, { WebpackPluginInstance } from 'webpack';
+import { BacktracePluginOptions } from './models/BacktracePluginOptions';
+
+export class BacktracePlugin implements WebpackPluginInstance {
+    private readonly _sourceMapProcessor: SourceProcessor;
+    private readonly _sourceMapUploader?: SourceMapUploader;
+
+    constructor(public readonly options?: BacktracePluginOptions) {
+        this._sourceMapProcessor = new SourceProcessor(options?.debugIdGenerator ?? new DebugIdGenerator());
+
+        this._sourceMapUploader =
+            options?.sourceMapUploader ?? (options?.uploadUrl ? new SourceMapUploader(options.uploadUrl) : undefined);
+    }
+
+    public apply(compiler: webpack.Compiler) {
+        compiler.hooks.afterEmit.tapPromise(BacktracePlugin.name, async (compilation) => {
+            const logger = compilation.getLogger(BacktracePlugin.name);
+            if (!compilation.outputOptions.path) {
+                logger.error('skipping everything because outputOptions.path is not set, a bug?');
```

Comment:
1. Skipping
2. "The outputOptions.path is not defined. Cannot generate source maps for not existing path" Isn't this description better?

Reply:
Generally, Webpack does not allow to configure it without an output path. So this theoretically should never happen. Even the type docs describe this as required, though for some reason, it is marked as optional.

![image](https://github.com/backtrace-labs/backtrace-javascript/assets/9085864/69378049-c115-43fe-8d83-eb40122f8553)